app/back-end/src/routes/workspace_aggregate_route.py

```diff
@@ -0,0 +1,359 @@
+# pylint: disable=import-error
+
+import os
+import csv
+from flask import Blueprint, request, jsonify
+from ast import literal_eval
+
+from src.setup.extensions import logger
+from src.utils.helpers import socketio_emit_to_user_session, is_number
+from src.utils.exceptions import UnexpectedError
+from src.constants import WORKSPACE_AGGREGATE_ROUTE, CONSOLE_FEEDBACK_EVENT, WORKSPACE_DIR
+
+
+workspace_aggregate_route_bp = Blueprint("workspace_aggregate_route", __name__)
+
+
+@workspace_aggregate_route_bp.route(
+    f"{WORKSPACE_AGGREGATE_ROUTE}/all/<path:relative_path>", methods=["GET"]
+)
+def get_workspace_aggregate_all(relative_path):
+    uuid = request.headers.get("uuid")
+    sid = request.headers.get("sid")
+
+    # Ensure the uuid header is present
+    if not uuid:
+        return jsonify({"error": "UUID header is missing"}), 400
+
+    # Ensure the sid header is present
+    if not sid:
+        return jsonify({"error": "SID header is missing"}), 400
+
+    # Emit a feedback to the user's console
+    socketio_emit_to_user_session(
+        CONSOLE_FEEDBACK_EVENT,
+        {"type": "info", "message": f"Calculating all file at '{relative_path}'..."},
+        uuid,
+        sid,
+    )
+
+    user_workspace_dir = os.path.join(WORKSPACE_DIR, uuid)
+    file_path = os.path.join(user_workspace_dir, relative_path)
+
+    columns_aggregation = request.args.get("columnsAggregation")
+    columns_aggregation = literal_eval(columns_aggregation)
+    header_actions = {
+        field: columns_aggregation[field]["action"] for field in columns_aggregation.keys()
+    }
+    header_values = {
+        field: (
+            float("inf")
+            if columns_aggregation[field]["action"] == "min"
+            else float("-inf") if columns_aggregation[field]["action"] == "max" else float(0)
+        )
+        for field in columns_aggregation.keys()
+    }
+    skipped_counts = {field: 0 for field in columns_aggregation.keys()}
+    counts = {field: 0 for field in columns_aggregation.keys()}
+
+    try:
+        with open(file_path, "r", encoding="utf-8") as file:
+            reader = csv.reader(file)
+            header = next(reader)
+
+            if header:
+                for row in reader:
+                    for field in columns_aggregation.keys():
+                        action = header_actions[field]
+                        value = row[header.index(field)]
```

Comment:
Need to check if the value is not null

Reply:
In this place it is not necessary cause the following lines indirectly do this function, but in other parts this check must be added since otherwise the application breaks

Reply:
Added check